File: src/com/eagle/programmar/CSharp/Terminals/CSharp_Number.java
// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Dec 18, 2010

package com.eagle.programmar.CSharp.Terminals;

import com.eagle.programmar.CSharp.CSharp_Expression;
import com.eagle.programmar.CSharp.CSharp_Generator;
import com.eagle.tokens.AbstractToken;
import com.eagle.tokens.terminals.TerminalNumberToken;

public class CSharp_Number extends TerminalNumberToken
{
	public CSharp_Number()
	{
		super("Ee", "LlFfDdUuMm", true, false, '?');
	}

	public static CSharp_Number createNumber(int value)
	{
		return generateNumber(Integer.toString(value), null);
	}
	
	public static CSharp_Number generateNumber(String value, AbstractToken source)
	{
		CSharp_Number num = new CSharp_Number();
		num.setValue(value);
		num.setTransformationSource(source);
		return num;
	}

	public static CSharp_Expression generateNumberExpression(String value, AbstractToken source)
	{
		return CSharp_Generator.wrapExpression(generateNumber(value, source));
	}
}

File: csrc/com/eagle/programmar/CSharp/Terminals/CSharp_Number.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Dec 18, 2010

namespace com.eagle.programmar.CSharp.Terminals
{
	using CSharp_Expression = com.eagle.programmar.CSharp.CSharp_Expression;
	using CSharp_Generator = com.eagle.programmar.CSharp.CSharp_Generator;
	using AbstractToken = com.eagle.tokens.AbstractToken;
	using TerminalNumberToken = com.eagle.tokens.terminals.TerminalNumberToken;

	public class CSharp_Number : TerminalNumberToken
	{
		public CSharp_Number() : base("Ee", "LlFfDdUuMm", true, false, '?')
		{
		}

		public static CSharp_Number generateNumber(string value, AbstractToken source)
		{
			CSharp_Number num = new CSharp_Number();
			num.setValue(value);
			num.setTransformationSource(source);
			return num;
		}

		public static CSharp_Expression generateNumberExpression(string value, AbstractToken source)
		{
			return CSharp_Generator.wrapExpression(generateNumber(value, source));
		}
	}

}
